src/corefx/System.Globalization.Native/locale.cpp

```diff
@@ -0,0 +1,373 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#include <stdint.h>
+#include <string.h>
+
+#include "localeInformationConstants.hpp"
+
+#include "unicode/dcfmtsym.h" //decimal
+#include "unicode/dtfmtsym.h" //date
+#include "unicode/ulocdata.h"
+
+/*
+Function:
+GetLocale
+
+Returns a locale given the locale name
+*/
+Locale GetLocale(const UChar* localeName, bool canonize = false)
```

Comment:
One thing I had considered is having sWindowsName in things like CompareInfo and CultureData actually be a logical byte* which pointed to ASCII data, which was the locale id tag.  Then, on the interop boundary, consume the byte* as a char* and avoid having to do all these conversions.

Interested in what you thought about an approach like this?